st2actions/st2actions/runners/pythonrunner.py

```diff
@@ -158,34 +168,61 @@ def run(self, action_parameters):
         if ACTION_OUTPUT_RESULT_DELIMITER in stdout:
             split = stdout.split(ACTION_OUTPUT_RESULT_DELIMITER)
             assert len(split) == 3
-            result = split[1].strip()
+            action_result = split[1].strip()
             stdout = split[0] + split[2]
         else:
-            result = None
+            action_result = None
 
         try:
-            result = json.loads(result)
+            action_result = json.loads(action_result)
         except:
             pass
+        action_status = None
+        if action_result and action_result != "None":
+            result = action_result['result']
```

Comment:
This could also throw KeyError in case `result` key is not present in the dictionary, right?